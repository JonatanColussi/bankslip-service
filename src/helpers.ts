import { APIGatewayProxyResult } from 'aws-lambda';

export function response(data: Record<string, any>, statusCode?: number) {
  return Promise.resolve({
    body: JSON.stringify(data),
    headers: {
      'Content-Type': 'Application/json',
    },
    statusCode: statusCode || 200,
  } as APIGatewayProxyResult);
}
