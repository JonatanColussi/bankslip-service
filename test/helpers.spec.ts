import { response } from 'helpers';

it('should return success response with any object', async () => {
  expect(await response({ foo: 'bar' })).toMatchSnapshot();
});

it('should return error response with any object', async () => {
  expect(await response({ message: 'any_error_message' }, 400)).toMatchSnapshot();
});
