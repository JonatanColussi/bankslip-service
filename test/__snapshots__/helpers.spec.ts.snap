// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return error response with any object 1`] = `
Object {
  "body": "{\\"message\\":\\"any_error_message\\"}",
  "headers": Object {
    "Content-Type": "Application/json",
  },
  "statusCode": 400,
}
`;

exports[`should return success response with any object 1`] = `
Object {
  "body": "{\\"foo\\":\\"bar\\"}",
  "headers": Object {
    "Content-Type": "Application/json",
  },
  "statusCode": 200,
}
`;
